src/Services/QueuedJobService.php

```diff
@@ -672,28 +703,59 @@ protected function initialiseJob(QueuedJobDescriptor $jobDescriptor)
      */
     protected function grabMutex(QueuedJobDescriptor $jobDescriptor)
     {
-        // write the status and determine if any rows were affected, for protection against a
-        // potential race condition where two or more processes init the same job at once.
-        // This deliberately does not use write() as that would always update LastEdited
-        // and thus the row would always be affected.
         try {
-            DB::query(sprintf(
-                'UPDATE "QueuedJobDescriptor" SET "JobStatus" = \'%s\' WHERE "ID" = %s'
-                . ' AND "JobFinished" IS NULL AND "JobStatus" NOT IN (%s)',
-                QueuedJob::STATUS_INIT,
-                $jobDescriptor->ID,
-                "'" . QueuedJob::STATUS_RUN . "', '" . QueuedJob::STATUS_COMPLETE . "', '"
-                . QueuedJob::STATUS_PAUSED . "', '" . QueuedJob::STATUS_CANCELLED . "'"
-            ));
+            // Start a transaction which will hold until we have a lock on this descriptor.
+            DB::get_conn()->withTransaction(function () use ($jobDescriptor) {
+                $query = 'SELECT `ID`, UUID() as `mutex`, `Worker` FROM `QueuedJobDescriptor`'
```

Comment:
`UUID()` function is MySQL specific and won't work in other databases, can we generate it on PHP side?